### PR TITLE
Applicant model updates

### PR DESCRIPTION
**Pull request by thomaslapadat**
- Updated the applicant model:
    - Added the new 'level' field
    - Updated the list of valid statuses
- Replaces updateApplicantStatus() with updateApplicantFields() in the applicant controller
    - Updated the check for valid status messages
    - PATCH can now update an applicant's level

### CODE REVIEW

File: src/controllers/applicant.ts [hunk 1]
```diff
@@ -22,31 +24,46 @@ export const createApplicant = async (req: Request, res: Response): Promise<void
...
+        'Pending',
+        'Screened: Accepted',
+        'Screened: Rejected',
+        'Scheduled for Interview',
+        'Final Decision: Accepted',
+        'Final Decision: Rejected',
+        'Archived: Rejected',
+    ];
+    const valid_level = ['Beginner', 'Intermediate', 'Advanced'];
+    const allowed_updates = ['id', 'status', 'level']; // id is passed in the body but is not an update
```
Comment (DConnorFong) [inline src/controllers/applicant.ts]: We should change the API route to PATCH applicant/:applicantId - I don't think we should open the opportunity for somebody to change an applicant ID that would be pretty chaotic.

**Review by DConnorFong — CHANGES_REQUESTED** [review]

File: src/controllers/applicant.ts (hunk 1 above)
Comment (thomaslapadat) [reply src/controllers/applicant.ts]: I've applied the requested changes

File: src/controllers/applicant.ts [hunk 2]
```diff
@@ -43,29 +44,63 @@ export const createApplicant = async (req: Request, res: Response): Promise<void
...
-        const my_applicant = await Applicant.findOne({ _id: req.body.id });
+export const updateApplicantFields = async (req: Request, res: Response): Promise<void> => {
+    const valid_status = [
+        'Pending',
+        'Screened: Accepted',
+        'Screened: Rejected',
+        'Scheduled for Interview',
+        'Final Decision: Accepted',
+        'Final Decision: Rejected',
+        'Archived: Rejected',
```
Comment (DConnorFong) [inline src/controllers/applicant.ts]: Sorry - one more small change, can we added a "Interviewed" and "Screened" state here too so we don't have to move directly from scheduled for interview to a decision (and same for after screening an applicant)

**Review by DConnorFong — CHANGES_REQUESTED** [review]

File: src/controllers/applicant.ts (hunk 2 above)
Comment (thomaslapadat) [reply src/controllers/applicant.ts]: I've added those states now in my most recent commit

**Review by DConnorFong — APPROVED** [review]

**Pull request merged**